### PR TITLE
RavenDB-21646 : fix flaky test

### PR DESCRIPTION
**Pull request by aviv86**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21646/SlowTests.Server.Replication.ReplicationCleanTombstones.CanBackupAndRestoreTombstonesWithSameIdoptions-DatabaseMode-Sharded

### Additional description

-  wait for backups to complete before starting the restore process
- after restore check the tombstones ids in the restored database, instead of checking in the original one

### Type of change

- Bug fix
- Tests stabilization

### How risky is the change?

- Low

### CODE REVIEW

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs [hunk 1]
```diff
@@ -879,5 +862,53 @@ public async Task CanBackupAndRestoreTombstonesWithSameId(Options options)
...
+                return await Backup.UpdateConfigAndRunBackupAsync(Server, config, store);
+            
+            var backupCompleted = await Sharding.Backup.WaitForBackupToComplete(store);
+            var backupTaskId = await Sharding.Backup.UpdateConfigurationAndRunBackupAsync(Server, store, config);
+            Assert.True(WaitHandle.WaitAll(backupCompleted, TimeSpan.FromSeconds(10)));
+
+            return backupTaskId;
+        }
+
+        private async Task<RestoreBackupConfiguration> RunBackupAndCreateRestoreConfiguration(DocumentStore store, RavenDatabaseMode mode, long backupTaskId, string backupPath)
```
Comment (karmeli87) [inline test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: does it make sense to move this to the backup class?

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs [hunk 2]
```diff
@@ -879,5 +862,53 @@ public async Task CanBackupAndRestoreTombstonesWithSameId(Options options)
...
+                var dirs = Directory.GetDirectories(backupPath);
+                var sharding = await Sharding.GetShardingConfigurationAsync(store);
+                var settings = Sharding.Backup.GenerateShardRestoreSettings(dirs, sharding);
+                configuration.ShardRestoreSettings = settings;
+            }
+
+            return configuration;
+        }
+
+        private async Task<DocumentsStorage> GetDocumentStorageFor(IDocumentStore store, RavenDatabaseMode mode, string databaseName, string id)
```
Comment (karmeli87) [inline test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: can we reuse `GetDocumentDatabaseInstanceForAsync`?

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs [hunk 3]
```diff
@@ -879,5 +862,53 @@ public async Task CanBackupAndRestoreTombstonesWithSameId(Options options)
                 }
             }
         }
+
+        private async Task<long> CreateAndRunBackup(IDocumentStore store, RavenDatabaseMode mode, string backupPath)
```
Comment (karmeli87) [inline test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: I think we should move this to the infra as well

Comment (karmeli87) [reply test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: `CreateAndRunBackupAsync`

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs (hunk 1 above)
Comment (karmeli87) [reply test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: `RunBackupAndCreateRestoreConfigurationAsync`

Comment (aviv86) [reply test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: done

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs (hunk 2 above)
Comment (aviv86) [reply test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: done

File: test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs (hunk 3 above)
Comment (aviv86) [reply test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs]: done

**Review by karmeli87 — APPROVED** [review]

**Pull request merged**